homeassistant/components/zha/device_entity.py

```diff
@@ -101,6 +101,19 @@ def device_state_attributes(self):
             # only do this on add to HA because it is static
             await self._async_init_battery_values()
 
+    def async_update_state_attribute(self, key, value):
+        """Update a single device state attribute."""
+        if key == 'battery_level':
+            if value is not None and value != -1:
```

Comment:
I'd suggest making sure it's a number to avoid a `TypeError` if it's not `None` but not a number.  You could also invert condition to return right away instead of having an if/else branch:

```python
from numbers import Number

if not isinstance(value, Number) or value == -1:
    return
```

Reply:
good suggestion. Thanks!